client/X11/xf_window.c

```diff
@@ -942,7 +944,11 @@ void xf_SetWindowVisibilityRects(xfContext* xfc, xfAppWindow* appWindow, RECTANG
 		xrects[i].height = rects[i].bottom - rects[i].top;
 	}
 
-	XShapeCombineRectangles(xfc->display, appWindow->handle, ShapeBounding, 0, 0, xrects, nrects, ShapeSet, 0);
+	/* Translate the server rail window offset to a local offset */
+	translatedWindowOffsetX = (appWindow->visibleOffsetX - appWindow->windowOffsetX);
+	translatedWindowOffsetY = (appWindow->visibleOffsetY - appWindow->windowOffsetY);
```

Comment:
I guess the non-negative case would be the one to be concerned about. Either way, I just do not think the logic here is always doing a translation to a local offset. The purpose of the "appWindow->localWindowOffsetCorr" variables was to handle cases where the "appWindow->windowOffset" variables are slightly negative - such as the case when you maximize a window. If they get large negative values when you slide the window off the left side of the screen then I think the overall logic is wrong(and I mean the existing logic beyond the scope of your commit) - as currently we are offseting anything negative into our coordinate system which does not support negative values when maybe we should just be truncating off the negative part. I will try to take a look at this sometime this week also.